File: AuraFit-BackEnd/src/main/java/app/aurafitbackend/Beans/Review.java
package app.aurafitbackend.Beans;

import app.aurafitbackend.Enums.Rating;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "reviews")
@NoArgsConstructor
@AllArgsConstructor
@Data
public class Review {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @ManyToOne
    private User user;
    @Enumerated(EnumType.STRING)
    private Rating rating;
    private String content;
    private LocalDateTime reviewDate;
    @ManyToOne
    private Product product;

}

File: aurafit-frontend/src/Models/Review.ts
import {User} from "./User.ts";
import {Product} from "./Product.ts";
import { Rating } from "./Enums/Rating.ts";

export class Review {
    id: number;
    user: User;
    rating: number;
    content: string;
    reviewDate: Date;
    product: Product;

    constructor(id: number, user: User, rating: number, content: string, reviewDate: Date, product: Product) {
        this.id = id;
        this.user = user;
        this.content = content;
        this.rating = rating;
        this.reviewDate = reviewDate;
        this.product = product;
    }
}